spinalcordtoolbox/deepseg/models.py

```diff
@@ -420,6 +429,49 @@ def install_default_models():
             install_model(name_model)
 
 
+def is_up_to_date(path_model):
+    """
+    Determine whether an on-disk model is up-to-date by comparing
+    the URL used to download the model with the latest mirrors.
+
+    :return: bool: whether the model is up-to-date
+    """
+    source_path = os.path.join(path_model, "source.json")
+    if not os.path.isfile(source_path):
+        logger.warning("Provenance file 'source.json' missing!")
+        return False  # NB: This will force a reinstall
+    with open(source_path, "r") as fp:
+        source_dict = json.load(fp)
+    model_name = source_dict["model_name"]
+    if model_name not in MODELS:
+        logger.warning(f"Model name '{model_name}' from source.json does not match model names in SCT source code.")
+        return False
+    expected_model_urls = MODELS[model_name]['url']
```

Comment:
Since we mutate this later with `expected_model_urls.pop`, it should be a copy (which should work whether it's a list or a dict):
```suggestion
    expected_model_urls = MODELS[model_name]['url'].copy()
```